## 1. Install Beberapa Library Terlebih Dahulu

In [1]:
# Install library nltk (natural language tool-kit)
pip install nltk

Note: you may need to restart the kernel to use updated packages.


In [2]:
# Install library WordCloud. Library ini untuk membuat visualisasi kumpulan-kumpulan teks
pip install WordCloud

Note: you may need to restart the kernel to use updated packages.


## 2. Import Library dan Eksplorasi Dataset Menggunakan Pandas

In [22]:
# Import Library
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
import re
import nltk
from nltk.tokenize import word_tokenize
from sklearn.metrics import confusion_matrix, classification_report

In [23]:
# Read dataset yang telah dicrawling (diambil) dari media sosial twitter dan menampilkan data sebanyak 17 data teratas
data_hepatitis = pd.read_csv("Data Crawling Hepatitis Akut/Tweet_Hepatitis_with_Label_TextBlob.csv")

data_hepatitis.head(17)

Waktu Tweet                   Username  \
0   2022-06-26 16:16:08  ⚕️base anak fk | open dm📩   
1   2022-06-26 14:02:24           billion store id   
2   2022-06-26 12:19:45                    lyndsay   
3   2022-06-26 12:19:42                    lyndsay   
4   2022-06-26 12:17:25                    lyndsay   
5   2022-06-26 11:20:51                   syaza 🇲🇾   
6   2022-06-26 11:17:15        jengkelin say no 🚫💉   
7   2022-06-26 11:08:07            rs pondok indah   
8   2022-06-26 11:07:32                          🦋   
9   2022-06-26 11:05:22                       nawa   
10  2022-06-26 10:20:14              ᗰᗩᗪᗩᗰ ᑎᘜᗴ̄ᕼᗴ̄   
11  2022-06-26 09:24:49           jangantungguhaus   
12  2022-06-26 09:12:36                  inisial d   
13  2022-06-26 08:36:42                hendi putra   
14  2022-06-26 08:18:06                       lina   
15  2022-06-26 07:23:35             pikiran rakyat   
16  2022-06-26 07:10:33                   oke zone   

                                                Tweet  Sentimen  
0   dok mau tanyaaa kalo misalnya tes hepatitis se...   Neutral  
1   terbaik teraman supplement kesehatan kapsul sp...  Positive  
2   ketua dpr puan maharani menyoroti sejumlah hal...  Positive  
3   puan yang menyoroti hepatitis dan berbagai mac...   Neutral  
4   puan menyoroti kasus hepatitis akut anak harga...  Positive  
5   ambil ujian darah doktor kata darah semua cant...  Positive  
6   vaksin gak pernah salah jadi katakan saja itu ...  Positive  
7   sempat ramai beberapa waktu lalu hepatitis mis...  Positive  
8                             mulan hepatitis gunung    Neutral  
9   byeeee semua bye gue udah stres minggu sama ke...   Neutral  
10   gila dan itu mmg benar banyak anak2 sudah vak...  Negative  
11   rilissehat patogen ini paling banyak ditemuka...   Neutral  
12   kalau perawatan kecantikan udah sering dibaha...  Positive  
13  genjot vaksinasi booster binda jateng vaksinas...  Positive  
14  hepatitis tidak terkait dengan vaksin vaksin 1...  Positive  
15  who peringatkan hepatitis misterius pada anak ...   Neutral  
16  hepatitis akut masih mengintai bunda ini cara ...  Positive

In [24]:
# Melihat ukuran seperti jumlah kolom dan baris dari DataFrame "data_hepatitis"
data_hepatitis.shape

(899, 4)

In [25]:
# Melihat tipe data dari DataFrame "data_hepatitis"
data_hepatitis.dtypes

Waktu Tweet    object
Username       object
Tweet          object
Sentimen       object
dtype: object

In [26]:
# Melihat apakah di dalam DataFrame "data_hepatitis" memiliki missing values/tidak
data_hepatitis.isna()

Waktu Tweet  Username  Tweet  Sentimen
0          False     False  False     False
1          False     False  False     False
2          False     False  False     False
3          False     False  False     False
4          False     False  False     False
..           ...       ...    ...       ...
894        False     False  False     False
895        False     False  False     False
896        False     False  False     False
897        False     False  False     False
898        False     False  False     False

[899 rows x 4 columns]

In [27]:
# Melihat informasi yang terkandung di dalam DataFrame "data_hepatitis"
data_hepatitis.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 899 entries, 0 to 898
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Waktu Tweet  899 non-null    object
 1   Username     899 non-null    object
 2   Tweet        899 non-null    object
 3   Sentimen     899 non-null    object
dtypes: object(4)
memory usage: 28.2+ KB


Berdasarkan output dari informasi yang ada di dalam "data_hepatitis" maka dapat diambil beberapa kesimpulan diantaranya adalah sebagai berikut:

1. Jumlah baris yang ada di dalam DataFrame "data_hepatitis" yakni sebanyak 899 data dengan 4 kolom.
2. Setiap kolom yang ada di dalam DataFrame "data_hepatitis" tidak memiliki missing values/nilai kosong.
3. Tipe data yang digunakan pada DataFrame "data_hepatitis" hanya object (string) saja.

In [28]:
"""
Melihat jumlah nilai sekaligus data-data apa saja yang ada 
di dalam kolom "Waktu Tweet", "Username", dan "Sentimen"
"""

print("--------------Jumlah Total Nilai Data Pada Kolom Waktu Tweet--------------")
print(data_hepatitis["Waktu Tweet"].value_counts(), "\n")

print("--------------Jumlah Total Nilai Data Pada Kolom Username--------------")
print(data_hepatitis["Username"].value_counts(), "\n")

print("--------------Jumlah Total Nilai Data Pada Kolom Sentimen--------------")
print(data_hepatitis["Sentimen"].value_counts())

--------------Jumlah Total Nilai Data Pada Kolom Waktu Tweet--------------
2022-06-19 05:55:41    2
2022-06-19 11:42:28    2
2022-06-19 11:42:29    2
2022-06-22 07:20:48    2
2022-06-24 02:05:59    2
                      ..
2022-06-24 02:28:17    1
2022-06-24 02:27:38    1
2022-06-24 02:26:50    1
2022-06-24 02:22:18    1
2022-06-17 01:45:06    1
Name: Waktu Tweet, Length: 894, dtype: int64 

--------------Jumlah Total Nilai Data Pada Kolom Username--------------
anissa billa                           14
yuni marisa                            10
sutansegeh                             10
putri                                  10
luis chaniago                           9
                                       ..
goodman                                 1
epidemiology job openings (epijobs)     1
mune 🌙                                  1
kalla jengking                          1
sunday🌏                                 1
Name: Username, Length: 514, dtype: int64 

--------------Jumlah Tota

Berdasarkan hasil output diatas maka dapat diambil contoh penjelasan dari dua kolom (Username dan Sentimen) yang diantaranya adalah sebagai berikut:


1. Pada kolom "Username", nilai yang sering muncul di dalam DataFrame "data_hepatitis" adalah username yang bernama annisa billa dengan nilai kemunculan sebanyak 14 kali yang kemudian disusul dengan username yang bernama yuni marisa yang memiliki nilai kemunculan sebanyak 10 kali, sedangkan username yang jarang terlihat pada DataFrame "data" adalah goodman, epidemiology job openings (epijobs), mune 🌙, kalla jengking, dan sunday🌏 yang memiliki nilai kemunculan hanya sebanyak satu kali saja dalam meng-tweet topik tentang hepatitis akut.
2. Pada kolom "Sentimen", nilai yang sering muncul di dalam kolom ini yaitu sentimen Positive dengan nilai kemunculan sebanyak 558 kali yang kemudian disusul oleh sentimen Neutral (Netral) sebanyak 243 kali, sedangkan sentimen Negative (Negatif) memiliki total nilai kemunculan sebanyak 98 kali di dalam DataFrame "data_hepatitis".

In [29]:
# Melihat nilai statistik deskriptif dari DataFrame "data_hepatitis" untuk semua kolom
data.describe(include="all")

Waktu Tweet      Username  \
count                   899           899   
unique                  894           514   
top     2022-06-19 05:55:41  anissa billa   
freq                      2            14   

                                                    Tweet  Sentimen  
count                                                 899       899  
unique                                                899         3  
top     dok mau tanyaaa kalo misalnya tes hepatitis se...  Positive  
freq                                                    1       558

Berdasarkan hasil diatas dapat diketahui nilai-nilai statistik deskriptif dari keempat kolom pada DataFrame "data_hepatitis". Di samping itu, karena keempat kolom ini bersifat kategorik bukan numerik maka nilai statistik deskriptif yang ditampilkan berbeda daripada nilai statistik deskriptif untuk kolom yang bersifat numerik karena nilai statistik deskriptif untuk kolom yang bersifat numerik adalah mean, std (standard deviasi), quartil 1 (25%), quartil 2/median (50%), quartil 3 (75%), min (minimum), dan max (maximum), sedangkan untuk kolom kategorik memiliki nilai statistik deskriptif yaitu nilai unique, top, dan freq.

Nilai unique merupakan total nilai/data yang berbeda di dalam 1 kolom. Nilai top merupakan nilai yang sering muncul di dalam kolom tersebut. Terakhir adalah nilai freq yang merupakan total nilai pada nilai top. Sebagai contoh nilai top pada kolom "Sentimen" adalah Positive yang memiliki total kemunculan di dalam kolom tersebut sebanyak 558 data. Hal tersebut sama seperti di codingan sebelumnya ketika saya ingin melihat jumlah nilai dari masing-masing kolom di dalam DataFrame "data_hepatitis". Berikut adalah penjelasan untuk masing-masing nilai statistik deskriptif dari kolom Waktu Tweet, Username, dan Sentimen berdasarkan hasil statistik deskriptif diatas:


1. Pada kolom "Waktu Tweet" memiliki nilai unique berjumlah 894 dengan nilai top/nilai yang sering terlihat adalah 2022-06-19 05:55:41 dengan nilai freq atau nilai kemunculan sebanyak 2 kali.
2. Kolom "Username" memiliki nilai top adalah anissa billa dengan jumlah nilai kemunculan sebanyak 14 kali. Selain itu, kolom "Username" memiliki 514 nilai data yang berbeda-beda.
3. Terakhir adalah kolom "Sentimen" yang memiliki nilai unique hanya 3 saja dan dari ketiga nilai tersebut terdapat nilai top/nilai yang sering muncul di dalam kolom "Sentimen" adalah sentimen Positive yang memiliki total nilai kemunculan (nilai freq) sebanyak 558 kali.